src/main/java/com/github/quiltservertools/ledger/mixin/FillCommandMixin.java

```diff
@@ -0,0 +1,34 @@
+package com.github.quiltservertools.ledger.mixin;
+
+import com.github.quiltservertools.ledger.callbacks.BlockChangeCallback;
+import com.github.quiltservertools.ledger.utility.Sources;
+import net.minecraft.block.BlockState;
+import net.minecraft.command.argument.BlockStateArgument;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.command.FillCommand;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(FillCommand.class)
+public class FillCommandMixin {
+    @Redirect(method = "execute", at = @At(value = "INVOKE", target = "Lnet/minecraft/command/argument/BlockStateArgument;setBlockState(Lnet/minecraft/server/world/ServerWorld;Lnet/minecraft/util/math/BlockPos;I)Z"))
```

Comment:
not sure but would it be better to inject just after `if (!((BlockStateArgument)blockStateArgument).setBlockState(serverWorld, blockPos, Block.NOTIFY_LISTENERS)) continue;` then use local capture to get the blockpos that have been changed? (i guess it would also need another inject to store the old state unless something still has the old state)

Reply:
Honestly I probably should have done setblock the same as fill, but I wrote it after and didn't think to go back and change it.

We would need at least two injects because we need the old state and as far as I can tell /fill's execute method doesn't store it anywhere. I chose not to go that route because that whole method is a little cryptic to me and I wanted to make sure only blocks that we're changed got logged and only logged once also I think the redirect way is a little clearer.